### ISSUE TITLE
multi_gpu_model ERROR: Incompatible shapes: [16,128] vs. [32,128]

### ISSUE THREAD

**Issue by cdj0311** [keras-team/keras#8976]
Hi, all:
### I have an error:
       InvalidArgumentError (see above for traceback): Incompatible shapes: [16,128] vs. [32,128]
	 [[Node: replica_0/sequential_1/lstm_1/while/mul_5 = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](replica_0/sequential_1/lstm_1/while/Identity_2, replica_0/sequential_1/lstm_1/while/mul_5/Enter)]]
	 [[Node: replica_0/sequential_1/dense_1/Sigmoid/_165 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_869_replica_0/sequential_1/dense_1/Sigmoid", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]
###  with follows code:
rom keras.utils import multi_gpu_model
from keras.preprocessing import sequence
from keras.models import Sequential
from keras.layers import Dense, Embedding
from keras.layers import LSTM
from keras.datasets import imdb
import os
"""
I have 8 gpus
"""
os.environ["CUDA_VISIBLE_DEVICES"] = "6,7"

model = Sequential()
model.add(Embedding(max_features, 128))
model.add(LSTM(128, dropout=0.2, recurrent_dropout=0.2))
model.add(Dense(1, activation='sigmoid'))
"""
multi_gpu_model
"""
model = multi_gpu_model(model, gpus=2)
model.compile(loss='binary_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])

print('Train...')
model.fit(x_train, y_train,
          batch_size=batch_size,
          epochs=2,
          validation_data=(x_test, y_test))

**Comment by npmhung** [comment]
Does this work on single GPU version?
I believe there's a mismatch between the max_features and the shape of x_train or x_test.
